### ISSUE TITLE
Is the train.log right?

### ISSUE THREAD

**Issue by zy486at189cn** [kenshohara/3D-ResNets-PyTorch#45]
@kenshohara 

First , it is a great job! 

I use the kinetics data to train a resnet-34-kinetics model. Every action have 50 MP4s. I train the model just like this:
python3 main.py --root_path ~/3D-ResNets-PyTorch --video_path kineticsJPG --annotation_path kineticsjson/kinetics.json --result_path results --dataset kinetics --model resnet --model_depth 34 --resnet_shortcut A --pretrain_path models/resnet-34-kinetics.pth --n_classes 400 --batch_size 30 --n_threads 4 --checkpoint 5
It success to create a mode "save_30.pth".
My train.log file looks like this:
epoch	loss	                            acc	                         lr
21	5.05777625165984	        0.050443081117927745	0.1
22	5.051979898375048	0.05023333857689686	0.1
23	5.037916659033671	0.05075769492947407	0.1
24	5.020105431997609	0.05448062503277227	0.1
25	5.006561265645096	0.05689266425462745	0.1
26	4.986441563412287	0.05615856536101935	0.1
27	4.982790104450624	0.05909496093545173	0.1
28	4.962981188055365	0.05883278275916313	0.1
29	4.957985667213246	0.06077290126369881	0.1
30	4.930375250401841	0.06276545540349221	0.1 
Is it OK?
When i use "video-classification-3d" to check the model ,
Command is:
python3 main.py --input ./input --video_root ./videos --output ./output.json --model pathto/save_30.pth --mode score

I find the result is poor.

Why?
Is the train.log right?
Do i train the model sufficiently?

**Comment by kenshohara** [comment]
using --pretrain_path initializes the last fc layer.
So, the training of the last fc may need more epochs.
If you want to use my pretrained model, you should just use resnet-34-kinetics.pth.
If you want to confirm the training loss and accuracy of my pretrained model,
you should use not --pretrain_path but --resume_path.

**Comment by zy486at189cn** [comment]
@kenshohara Thanks!
But When I use --resume_path,just like this:
python3 main.py --root_path ~/3D-ResNets-PyTorch --video_path kineticsJPG --annotation_path kineticsjson/kinetics.json --result_path results --dataset kinetics --model resnet --model_depth 34 --resnet_shortcut A --resume_path models/resnet-34-kinetics.pth  --n_classes 400 --batch_size 30 --n_threads 4 --checkpoint 5
The main.py cannot train the model!
The output looks like this:
  (module): ResNet(
    (conv1): Conv3d(3, 64, kernel_size=(7, 7, 7), stride=(1, 2, 2), padding=(3, 3, 3), bias=False)
    (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
    (relu): ReLU(inplace)
    (maxpool): MaxPool3d(kernel_size=(3, 3, 3), stride=2, padding=1, dilation=1, ceil_mode=False)
    (layer1): Sequential(
      (0): BasicBlock(
        (conv1): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
      )
      (1): BasicBlock(
        (conv1): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
      )
      (2): BasicBlock(
        (conv1): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(64, 64, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(64, eps=1e-05, momentum=0.1, affine=True)
      )
    )
    (layer2): Sequential(
      (0): BasicBlock(
        (conv1): Conv3d(64, 128, kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
      )
      (1): BasicBlock(
        (conv1): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
      )
      (2): BasicBlock(
        (conv1): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
      )
      (3): BasicBlock(
        (conv1): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(128, 128, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(128, eps=1e-05, momentum=0.1, affine=True)
      )
    )
    (layer3): Sequential(
      (0): BasicBlock(
        (conv1): Conv3d(128, 256, kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
      (1): BasicBlock(
        (conv1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
      (2): BasicBlock(
        (conv1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
      (3): BasicBlock(
        (conv1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
      (4): BasicBlock(
        (conv1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
      (5): BasicBlock(
        (conv1): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(256, 256, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(256, eps=1e-05, momentum=0.1, affine=True)
      )
    )
    (layer4): Sequential(
      (0): BasicBlock(
        (conv1): Conv3d(256, 512, kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(512, 512, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
      )
      (1): BasicBlock(
        (conv1): Conv3d(512, 512, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(512, 512, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
      )
      (2): BasicBlock(
        (conv1): Conv3d(512, 512, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn1): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
        (relu): ReLU(inplace)
        (conv2): Conv3d(512, 512, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
        (bn2): BatchNorm3d(512, eps=1e-05, momentum=0.1, affine=True)
      )
    )
    (avgpool): AvgPool3d(kernel_size=(1, 4, 4), stride=1, padding=0, ceil_mode=False, count_include_pad=True)
    (fc): Linear(in_features=512, out_features=400, bias=True)
  )
)
dataset loading [0/19906]
dataset loading [1000/19906]
dataset loading [2000/19906]
dataset loading [3000/19906]
dataset loading [4000/19906]
dataset loading [5000/19906]
dataset loading [6000/19906]
dataset loading [7000/19906]
dataset loading [8000/19906]
dataset loading [9000/19906]
dataset loading [10000/19906]
dataset loading [11000/19906]
dataset loading [12000/19906]
dataset loading [13000/19906]
dataset loading [14000/19906]
dataset loading [15000/19906]
dataset loading [16000/19906]
dataset loading [17000/19906]
dataset loading [18000/19906]
dataset loading [19000/19906]
dataset loading [0/3]
loading checkpoint ~/3D-ResNets-PyTorch/models/resnet-34-kinetics.pth
run

the program stop!
Why?The model resnet-34-kinetics.pth cannot be used as pretrain model?

**Comment by zy486at189cn** [comment]
@kenshohara Very Thanks! 

I use the resnet-34-kinetics.pth as the resume model,it can  confirm the training.Command just like this:
 python3 main.py --root_path ~/3D-ResNets-PyTorch --video_path kineticsJPG --annotation_path kineticsjson/kinetics.json --result_path results --dataset kinetics --model resnet --model_depth 34 --resnet_shortcut A --resume_path models/resnet-34-kinetics.pth  --n_classes 400 --batch_size 30 --n_threads 4 --checkpoint 5

When I use the trained model to detect MP4s,it can detect the MP4s right. such as cooking egg.
But the train.log looks like this:
epoch	loss	acc	lr
251	3.0146184289757327	0.3498505584395155	0.00010000000000000003
252	2.9790674948516047	0.3603376854910597	0.00010000000000000003
253	2.9866867607000285	0.3545697656127104	0.00010000000000000003
254	2.9517760721602415	0.36495202139373917	0.00010000000000000003
255	2.934799548907158	0.3653190708405432	0.00010000000000000003
256	2.933299742612108	0.36830790205023334	0.00010000000000000003
257	2.899572157721034	0.3686749514970374	0.00010000000000000003
258	2.9151977803772007	0.3697236642021918	0.00010000000000000003
259	2.8628139520602836	0.379843741806932	0.00010000000000000003
260	2.886826454439834	0.3741806931990981	0.00010000000000000003
261	2.8730955221789127	0.3779036233023963	0.00010000000000000003
262	2.8713922783022165	0.37989617744218973	0.00010000000000000003
263	2.8699963414220853	0.37328928739971684	1.0000000000000004e-05
264	2.865433342598961	0.3804205337947669	1.0000000000000004e-05
265	2.874772745988758	0.37680247496198416	1.0000000000000004e-05
266	2.850167988643193	0.37900477164280844	1.0000000000000004e-05
267	2.850641434571832	0.3815741177704368	1.0000000000000004e-05
268	2.853663224027801	0.38204603848775626	1.0000000000000004e-05

The final acc is 0.38204603848775626! Is the train.log right?

**Comment by slighting666** [comment]
@kenshohara @zy486at189cn 
Hi, I got the accuracy of val.log, but I want to calculate the accuracy of video level, where can I get the val.json file and how to use eval_ucf101.py?

**Comment by YTHmamba** [comment]
> 

@slighting666 I have the same question as you, there seems to be no eval_UCf101.py file in the code in the master branch
